pkg/registry/registry.go

```diff
@@ -292,11 +295,17 @@ func (r *registry) getSubjects() []rbac.Subject {
 // newDeployment() returns a Deployment object that can be used to bring up a
 // registry deployment
 func (r *registry) newDeployment(registryCommand []string, needServiceAccount bool) *apps.Deployment {
-	return new(builders.DeploymentBuilder).
+	builder := new(builders.DeploymentBuilder).
 		WithMeta(r.key.Name, r.key.Namespace).
-		WithOwnerLabel(r.key.Name, r.key.Namespace).
-		WithSpec(1, r.getLabel(), r.newPodTemplateSpec(registryCommand, needServiceAccount)).
-		Deployment()
+		WithSpec(1, r.getLabel(), r.newPodTemplateSpec(registryCommand, needServiceAccount))
+	if r.owner == v1.OperatorSourceKind {
+		return builder.WithOpsrcOwnerLabel(r.key.Name, r.key.Namespace).
+			Deployment()
+	} else if r.owner == v2.CatalogSourceConfigKind {
+		return builder.WithCscOwnerLabel(r.key.Name, r.key.Namespace).
+			Deployment()
```

Comment:
You are again defeating the purpose of a builder. Just add the labels in the `if` statements.

This applies to every instance you are doing this.